flow-server/src/main/java/com/vaadin/flow/server/communication/IndexHtmlRequestHandler.java

```diff
@@ -358,6 +367,59 @@ private void addDevTools(Document indexDocument,
         indexDocument.body().appendChild(new Element("vaadin-dev-tools"));
     }
 
+    static boolean isAllowedDevToolsHost(AbstractConfiguration configuration,
+            VaadinRequest request) {
+        String remoteAddress = request.getRemoteAddr();
+        String hostsAllowed = configuration
+                .getStringProperty(InitParameters.HOSTS_ALLOWED, null);
+
+        if (!isAllowedDevToolsHost(remoteAddress, hostsAllowed)) {
+            return false;
+        }
+        String forwardedFor = request.getHeader("X-Forwarded-For");
+        if (forwardedFor != null) {
+            if (forwardedFor.contains(",")) {
+                // X-Forwarded-For: <client>, <proxy1>, <proxy2>
+                // Elements are comma-separated, with optional whitespace
+                // surrounding the commas.
+                forwardedFor = forwardedFor.split(",")[0];
+            }
+            if (!isAllowedDevToolsHost(forwardedFor.trim(), hostsAllowed)) {
+                return false;
+            }
+        }
+
+        return true;
+
+    }
+
+    private static boolean isAllowedDevToolsHost(String remoteAddress,
+            String hostsAllowed) {
+        if (remoteAddress == null) {
+            // No remote address available so we cannot check...
+            return false;
+        }
+        // Always allow localhost
+        try {
+            InetAddress inetAddress = InetAddress.getByName(remoteAddress);
+            if (inetAddress.isLoopbackAddress()) {
+                return true;
+            }
+        } catch (Exception e) {
+            getLogger().debug(
+                    "Unable to resolve remote address: '{}', so we are preventing the web socket connection",
+                    remoteAddress, e);
+            return false;
+        }
+
+        if (hostsAllowed != null) {
+            // Allowed hosts set
+            return FilenameUtils.wildcardMatch(remoteAddress, hostsAllowed);
```

Comment:
Match easier than with regexp :+1: 
nit: I wonder if we should consider allowing `hostsAllowed` to provide multiple comma separated patterns

Reply:
Good idea, but I am not sure. 

And then I guess we need this code:
```suggestion
    String[] allowedHosts = hostsAllowed.split(",");

    for (String allowedHost : allowedHosts) {
        if (FilenameUtils.wildcardMatch(remoteAddress, allowedHost.trim())) {
            return true;
        }
    }
```

And then adding some covering tests... @Artur- I need you once more :D :( Do we wanna support that or wildcard shall be enough?

Reply:
An example from a setup with a reverse-proxy (running on docker) in front of an app, connection from a local and a remote machine

Accessing from local
```
browser URL: http://localhost:9090
remote addr: 172.17.0.3 (IP from docker network)
forwarded-for: 172.17.0.1 (IP from docker network)
```

Accessing from another laptop on the same LAN
```
browser URL: http://192.168.1.99:9090
remote addr: 172.17.0.3 (IP from docker network)
forwarded-for: 192.168.1.23 
```

To make it work, I would need to set the allowed list to something like `172.17.0.*,192.168.1.23`